tests/xcm-simulator/src/mocks/mod.rs

```diff
@@ -0,0 +1,192 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+pub(crate) mod msg_queue;
+pub(crate) mod parachain;
+pub(crate) mod relay_chain;
+
+use frame_support::traits::{Currency, OnFinalize, OnInitialize};
+use polkadot_parachain::primitives::{Id as ParaId, Sibling};
+use sp_runtime::traits::{AccountIdConversion, Get};
+use xcm::latest::prelude::*;
+use xcm_builder::{Account32Hash, SiblingParachainConvertsVia};
+use xcm_executor::traits::Convert;
+use xcm_simulator::{decl_test_network, decl_test_parachain, decl_test_relay_chain};
+
+pub const ALICE: sp_runtime::AccountId32 = sp_runtime::AccountId32::new([0xFAu8; 32]);
+pub const INITIAL_BALANCE: u128 = 1_000_000_000_000_000_000_000_000;
+pub const DAPP_STAKER_REWARD_PER_BLOCK: parachain::Balance = 1_000;
+pub const DAPP_STAKER_DEV_PER_BLOCK: parachain::Balance = 250;
+
+decl_test_parachain! {
+    pub struct ParaA {
+        Runtime = parachain::Runtime,
+        XcmpMessageHandler = parachain::MsgQueue,
+        DmpMessageHandler = parachain::MsgQueue,
+        new_ext = para_ext(1),
+    }
+}
+
+decl_test_parachain! {
+    pub struct ParaB {
+        Runtime = parachain::Runtime,
+        XcmpMessageHandler = parachain::MsgQueue,
+        DmpMessageHandler = parachain::MsgQueue,
+        new_ext = para_ext(2),
+    }
+}
+
+decl_test_relay_chain! {
+    pub struct Relay {
+        Runtime = relay_chain::Runtime,
+        XcmConfig = relay_chain::XcmConfig,
+        new_ext = relay_ext(),
+    }
+}
+
+decl_test_network! {
+    pub struct MockNet {
+        relay_chain = Relay,
+        parachains = vec![
+            (1, ParaA),
+            (2, ParaB),
+        ],
+    }
+}
+
+pub type RelayChainPalletXcm = pallet_xcm::Pallet<relay_chain::Runtime>;
+pub type ParachainPalletXcm = pallet_xcm::Pallet<parachain::Runtime>;
+pub type ParachainXcAssetConfig = pallet_xc_asset_config::Pallet<parachain::Runtime>;
+
+/// Derive parachain sovereign account on relay chain, from parachain Id
+pub fn para_account_id_on_relay(id: u32) -> relay_chain::AccountId {
+    ParaId::from(id).into_account_truncating()
+}
+
+/// Derive parachain sovereign account on a sibling parachain, from parachain Id
+pub fn sibling_para_account_id(id: u32) -> parachain::AccountId {
+    SiblingParachainConvertsVia::<Sibling, parachain::AccountId>::convert_ref(MultiLocation::new(
+        1,
+        X1(Parachain(id)),
+    ))
+    .unwrap()
+}
+
+/// Derive account32 hash for parachain Id
+/// TODO: improve or change this later
```

Comment:
probably better drop a few words about what needs improvement

Reply:
Removed this function since this week I learned that we'll use something different in the near future.